README.md

```diff
@@ -2,102 +2,78 @@
-    * A finalizer is added to referenced PV instance to preserve the PV. Attacher needs information from the PV to detach the volume.
-    * CSI `ControllerPublishVolume` is called.
-    * `AttachmentMetadata` is saved to `VolumeAttachment`.
-    * On any error, the `VolumeAttachment` is re-queued with exponential backoff.
-  * `VolumeAttachment` with `DeletionTimestamp`:
-    * CSI `ControllerUnpublishVolume` is called.
-    * A finalizer is removed from `VolumeAttachment`. At this point, the API server is going to delete this instance and "deleted `VolumeAttachment`" event will be received.
+## Overview
+The external-attacher is an external controller that monitors `VolumeAttachment` objects created by controller-manager and attaches/detaches volumes to/from nodes (i.e. calls `ControllerPublish`/`ControllerUnpublish`. Full design can be found at Kubernetes proposal at [container-storage-interface.md](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md)
 
-* Processes deleted `VolumeAttachment` instances:
-  * Pokes PV queue with name of the detached PV. This triggers removal of finalizer on PV, if needed.
+## Compatibility
 
-* Processes added/updated PV to remove finalizer on PVs:
-  * Ignore PVs that don't have DeletionTimestamp.
-  * Checks that the PV is not used by any `VolumeAttachment` instance.
-  * Removes Attacher's finalizer on the PV if so.
-  * On any error, the PV is re-queued with exponential backoff.
+This information reflects the head of this branch.
 
-#### Concurrency
+| Compatible with CSI Version                                                                | Container Image                     | Min K8s Version |
+| ------------------------------------------------------------------------------------------ | ------------------------------------| --------------- |
+| [CSI Spec v0.3.0](https://github.com/container-storage-interface/spec/releases/tag/v0.3.0) | quay.io/k8scsi/csi-attacher:v0.2.0  | 1.10            |
+| [CSI Spec v0.3.0](https://github.com/container-storage-interface/spec/releases/tag/v0.3.0) | quay.io/k8scsi/csi-attacher:v0.4.2  | 1.12            |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | quay.io/k8scsi/csi-attacher:v1.0.1  | 1.13            |
 
-Both PV queue and `VolumeAttachment` queue run in parallel. To ensure that removal of PV finalizers work without races:
+## Usage
 
-* The controller attaches PVs only when the PV has no DeletionTimestamp and has attacher's finalizer.
-* The controller removes finalizer only from PVs that have DeletionTimestamp.
+It is necessary to create a new service account and give it enough privileges to run the external-attacher. We provide one omnipotent yaml file that creates everything that's necessary, however it should be split into multiple files in production.
```

Comment:
Looking at the yaml, it only contains attacher and the mock driver.  What could you split into more files?

(Also I think the Service is unnecessary)

Reply:
Right, RBAC was cut out of there some time ago. I added a note about RBAC there.